### ISSUE TITLE
Null pointer exception in multipart.clj finish-up

### ISSUE THREAD

**Issue by josf** [juxt/yada#86]
In the :part and :in-partial parts of finish-up, there is a small bug (AFAICT):

`(if-let [e (+ (last positions) (:delimiter-size m))]`

I get a null pointer here. I'm assuming it is because `(last positions)` is returning nil. In any case, this will always either throw an exception or return true (if `+` succeeds). Probably want to test for truth before adding.

**Comment by malcolmsparks** [comment]
I can see how this might happen. Do you happen to have a failing test or some input data that causes the stack-trace that you can share? Even the size of the input data and boundary positions would be useful.

**Comment by josf** [comment]
I'll try to write this up as a test, or at least dump out the content. Just playing around on my app, I have determined that the error only occurs on files bigger than about 75 KB.

**Comment by mccraigmccraig** [comment]
set `{:raw-stream? true}` on your `aleph.http/start-server` opts and this problem goes away

https://github.com/ztellman/aleph/blob/master/src/aleph/http.clj#L32

and it only took about 18h of furious debugging to figure this out

**Comment by malcolmsparks** [comment]
@josf Does this clear up your problem? If so let me know and let's close this

**Comment by josf** [comment]
@malcolmsparks Yes, this works now! Thank you, @mccraigmccraig, for the furious debugging. I feel your pain.

One last thing:  there still seems to be a logical error in this `if-let` condition, which can never evaluate to anything other than true, since the call to `+` will never return a non-truthy value. 

Thanks again!

**Comment by malcolmsparks** [comment]
I'll keep this open to remind me to revisit the if-let condition

**Comment by mccraigmccraig** [comment]
yep, that's what this commit i did while debugging prevents (and thus reveals the underlying exception)

https://github.com/juxt/yada/commit/63872b9eb78136cb9a62e3c9640952b1b3c8443a

**Comment by mantree** [comment]
Encountered this NPE when negative testing my multi-part consumer implementation. To recreate throw an exception out of an implementation of start-partial.

**Comment by malcolmsparks** [comment]
Thanks for this Tom. Will look into it!
